gdc_client/upload/client.py

```diff
@@ -554,8 +542,9 @@ def upload_parts(self):
         if self.total_parts == 0:
```

Comment:
Can we do something better for the sake of reporting totals and completed when upload finishes. We can add another variable, say `previously_uploaded` and increment that instead of decrementing the `total_parts`, that way `total_parts` always remains the same and makes sensible reports in scenarios where the user doe a `resume` for a previously in-progress upload. This line becomes something like this
``` python
if self.total_parts == previously_uploaded:
    return
```
We will have to update this to reflect previously uploaded. https://github.com/NCI-GDC/gdc-client/blob/251dcb4ca104bfcf4d2ca4ea1b4dc761d8ff660f/gdc_client/upload/client.py#L584-L588

Reply:
Ok, so I added a variable called previously_uploaded, which keeps track of how many parts have been already uploaded in upload_parts. I then check if previously_uploaded == part_amount (if all parts have already been uploaded, just return). only after we get past that check do I set self.total_parts, which now represents the number of actual parts that are going to be attempted to be uploaded, if that makes sense.